File: app/src/main/java/com/example/iteventscftcheck_in/db/DataDao.java
package com.example.iteventscftcheck_in.db;


import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.iteventscftcheck_in.db.model.EventsModel;
import com.example.iteventscftcheck_in.db.model.ParticipantModel;

import java.util.List;

@Dao
public interface DataDao {
    @Insert
    void insertEvents(EventsModel eventsModel);

    @Delete
    void delete(EventsModel eventsModel);

    @Query("SELECT * FROM EventsModel")
    List<EventsModel> getAllData();

    @Query("SELECT * FROM EventsModel WHERE id LIKE :id")
    List<EventsModel> getDataById(int id);

    @Insert
    void insertParticipant(ParticipantModel participantModel);

    @Query("SELECT * FROM ParticipantModel")
    List<ParticipantModel> getAllDataParticipant();

    @Query("SELECT * FROM ParticipantModel WHERE id LIKE :id")
    List<ParticipantModel> getParticipantById(int id);
}

File: app/src/main/java/com/example/iteventscftcheck_in/db/DataDao.kt
package com.example.iteventscftcheck_in.db


import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

import com.example.iteventscftcheck_in.db.model.EventsModel
import com.example.iteventscftcheck_in.db.model.ParticipantModel

@Dao
interface DataDao {

    @get:Query("SELECT * FROM EventsModel")
    val allData: List<EventsModel>

    @get:Query("SELECT * FROM ParticipantModel")
    val allDataParticipant: List<ParticipantModel>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertEvents(eventsModel: EventsModel)

    @Query("SELECT * FROM EventsModel WHERE id LIKE :id")
    fun getDataById(id: Int): List<EventsModel>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertParticipant(participantModel: ParticipantModel)

    @Update
    fun update(participantModel: ParticipantModel)

    @Query("SELECT * FROM ParticipantModel WHERE id LIKE :id")
    fun getParticipantById(id: Int): ParticipantModel
}
